DataFormats/TrackingRecHitSoA/interface/TrackingRecHitsSoA.h

```diff
@@ -37,19 +24,36 @@ struct TrackingRecHitSoA {
                       SOA_COLUMN(int16_t, clusterSizeX),
                       SOA_COLUMN(int16_t, clusterSizeY),
                       SOA_COLUMN(uint16_t, detectorIndex),
-                      SOA_SCALAR(int32_t, offsetBPIX2),
-                      SOA_COLUMN(PhiBinnerStorageType, phiBinnerStorage),
-                      SOA_SCALAR(HitModuleStartArray, hitsModuleStart),
-                      SOA_SCALAR(HitLayerStartArray, hitsLayerStart),
-                      SOA_SCALAR(AverageGeometry, averageGeometry),
-                      SOA_SCALAR(PhiBinner, phiBinner));
-};
-
-template <typename TrackerTraits>
-using TrackingRecHitLayout = typename TrackingRecHitSoA<TrackerTraits>::template Layout<>;
-template <typename TrackerTraits>
-using TrackingRecHitSoAView = typename TrackingRecHitSoA<TrackerTraits>::template Layout<>::View;
-template <typename TrackerTraits>
-using TrackingRecHitSoAConstView = typename TrackingRecHitSoA<TrackerTraits>::template Layout<>::ConstView;
+                      SOA_SCALAR(int32_t, offsetBPIX2));
+
+  GENERATE_SOA_LAYOUT(HitModulesLayout, SOA_COLUMN(uint32_t, moduleStart));
+
+#ifdef ONLY_TRIPLETS_IN_HOLE
```

Comment:
You could generate the SoA, View, etc. data formats in any case, so that one does not need to rebuild all the dependencies to enable `ONLY_TRIPLETS_IN_HOLE`.

Reply:
In case, maybe leave a comment about `ONLY_TRIPLETS_IN_HOLE` so it's easier to find it later on.